fuel-vm/src/checked_transaction.rs

```diff
@@ -1146,20 +1363,21 @@ mod tests {
 
     fn is_valid_max_fee<Tx>(
         tx: &Tx,
+        gas_costs: &GasCosts,
         fee_params: &FeeParameters,
         base_asset_id: &AssetId,
     ) -> Result<bool, CheckError>
     where
         Tx: Chargeable + field::Inputs + field::Outputs,
     {
         let available_balances =
-            balances::initial_free_balances(tx, fee_params, base_asset_id)?;
+            balances::initial_free_balances(tx, gas_costs, fee_params, base_asset_id)?;
         // cant overflow as metered bytes * gas_per_byte < u64::MAX
-        let bytes = (tx.metered_bytes_size() as u128)
+        let fee = tx.metered_bytes_size() as u128
             * fee_params.gas_per_byte as u128
             * tx.price() as u128;
         let gas = tx.limit() as u128 * tx.price() as u128;
-        let total = bytes + gas;
+        let total = fee + gas;
```

Comment:
`max_fee` it is `min_fee` + (`GasLimit` converted into the price). So you need to update it in the same way as for `is_valid_min_fee`=)

Reply:
If I understand correctly:
```
max_fee = (min_fee + gas_limit) * gas_price
```

That would mean that `max_fee` is always greater than or equal to `min_fee`. 

After making that change, some tests fail, such as:
```rust
    #[test]
    fn base_fee_gas_limit_less_than_gas_used_by_predicates() {
        let metered_bytes = 5;
        let gas_used_by_signature_checks = 12;
        let gas_used_by_predicates = 8;
        let gas_limit = 7;
        let gas_price = 11;

        let fee = TransactionFee::checked_from_values(
            &PARAMS,
            metered_bytes,
            gas_used_by_signature_checks,
            gas_used_by_predicates,
            gas_limit,
            gas_price,
        )
        .expect("failed to calculate fee");

        assert!(fee.min_fee > fee.max_fee);
    }
 ``` 
where we test that the `min_fee` is greater than `max_fee`. 

Is there a specification for min/max fee?